### PR TITLE
[C# SDK] Fix PromptOptions<T> ambiguous constructor issue

### PR DESCRIPTION
**Pull request by ejadib**

### CODE REVIEW

File: CSharp/Library/Microsoft.Bot.Builder/Dialogs/PromptDialog.cs [hunk 1]
```diff
@@ -82,42 +82,113 @@ public enum PromptStyle
...
+        /// Default retry prompt that is used if <see cref="Retry"/> is null.
+        /// </summary>
+        string DefaultRetry { get; set; }
+
+        /// <summary>
+        /// Default retry speak that is used if <see cref="RetrySpeak"/> is null.
+        /// </summary>
+        string DefaultRetrySpeak { get; set; }
+
+        IPromptRecognizer Recognizer { get; }
```
Comment (msft-shahins) [inline CSharp/Library/Microsoft.Bot.Builder/Dialogs/PromptDialog.cs]: >Recognizer [](start = 26, length = 10)

missing documentation

File: CSharp/Library/Microsoft.Bot.Builder/Dialogs/PromptDialog.cs [hunk 2]
```diff
@@ -82,42 +82,113 @@ public enum PromptStyle
         None
     }
 
+    public interface IPromptOptions<T>
```
Comment (msft-shahins) [inline CSharp/Library/Microsoft.Bot.Builder/Dialogs/PromptDialog.cs]: >IPromptOptions [](start = 21, length = 14)

documentation missing

File: CSharp/Library/Microsoft.Bot.Builder/Dialogs/PromptDialog.cs [hunk 3]
```diff
@@ -206,9 +376,14 @@ public PromptOptions(string prompt, string retry = null, string tooManyAttempts
         /// <param name="speak"> The Speak tag (SSML markup for text to speech).</param>
         /// <param name="retrySpeak"> What to display on retry Speak (SSML markup for text to speech).</param>
         /// <param name="recognizer"> Entity Recognizer to parse the message content.</param>
-        public PromptOptions(string prompt, string retry = null, string tooManyAttempts = null, IReadOnlyDictionary<T, IReadOnlyList<T>> choices = null, int attempts = 3, PromptStyler promptStyler = null, IReadOnlyList<string> descriptions = null, string speak = null, string retrySpeak = null, IPromptRecognizer recognizer = null)
+        public PromptOptionsWithSynonyms(string prompt, string retry = null, string tooManyAttempts = null, IReadOnlyDictionary<T, IReadOnlyList<T>> choices = null, int attempts = 3, PromptStyler promptStyler = null, IReadOnlyList<string> descriptions = null, string speak = null, string retrySpeak = null, IPromptRecognizer recognizer = null)
         {
-            SetField.NotNull(out this.Prompt, nameof(this.Prompt), prompt);
```
Comment (msft-shahins) [inline CSharp/Library/Microsoft.Bot.Builder/Dialogs/PromptDialog.cs]: >NotNull [](start = 21, length = 7)

use SetField.NotNull(...) to be consistent

Comment (ejadib) [inline CSharp/Library/Microsoft.Bot.Builder/Dialogs/PromptDialog.cs]: You can't use SetField.NotNull with a property.

File: CSharp/Library/Microsoft.Bot.Builder/Dialogs/PromptDialog.cs (hunk 2 above)
Comment (ejadib) [inline CSharp/Library/Microsoft.Bot.Builder/Dialogs/PromptDialog.cs]: done

File: CSharp/Library/Microsoft.Bot.Builder/Dialogs/PromptDialog.cs (hunk 1 above)
Comment (ejadib) [inline CSharp/Library/Microsoft.Bot.Builder/Dialogs/PromptDialog.cs]: done

**Pull request merged**